proxy_manager.go

```diff
@@ -222,3 +222,68 @@ func (m *proxyManager) getFlakeIDGeneratorConfig(name string) FlakeIDGeneratorCo
 func makeProxyName(serviceName string, objectName string) string {
 	return fmt.Sprintf("%s%s", serviceName, objectName)
 }
+
+func destroyProxies(ctx context.Context, proxyArgs ...interface{}) error {
+	if proxyArgs == nil {
+		return fmt.Errorf("given proxy argument is nil")
+	}
+	errHandler := func(dsName string, i int, err error) error {
+		return fmt.Errorf("given %s proxy cannot be destroyed, last index: %d, %w", dsName, i, err)
+	}
+	for i, p := range proxyArgs {
+		if p == nil {
+			return fmt.Errorf("given %d. proxy argument is nil", i)
+		}
+		switch ds := p.(type) {
```

Comment:
Do we need this type switch? I am not sure if we need to know the actual type. What about using a type assertion to an interface like
```
`type Destroyer interface {
    Destroy(...)...
    }
```